### DISCUSSION TITLE
make launch and make PythonAPI error

### DISCUSSION THREAD

**Discussion by arsalan0302** [carla-simulator/carla#5756] [Ideas]
Hello!
I am facing an issue which are sgiven below, I have tried to solve it but could not reach  the specific path.
Please, guide us step by step as much as possible
![Capture](https://user-images.githubusercontent.com/112169068/190105678-7d5ec37b-1f98-4a02-88c7-5768a60b4864.PNG)
Any Suggestions?

**Comment by RaviBeagle** [comment]
On Windows you might have to use Visual Studio. I dont know if this works with MINGW

**Reply by arsalan0302** [reply]
> On Windows you might have to use Visual Studio. I dont know if this works with MINGW

Yes, I heve tried also with x64 prompt VS 17 but same error occurs

**Reply by arsalan0302** [reply]
> On Windows you might have to use Visual Studio. I dont know if this works with MINGW

@RaviBeagle Now here I faced new error.
 
![Capturess](https://user-images.githubusercontent.com/112169068/190667113-a763aba0-9747-4da7-8e8c-4d3b26950e5d.PNG)

if you know the solution please let me know

**Reply by ishaan95** [reply]
Can you delete the Build directory and run make PythonAPI then make launch?

**Reply by arsalan0302** [reply]
> Can you delete the Build directory and run make PythonAPI then make launch?

@ishaan95 I solved the problem, but when i run make launch it shows (doesn't detect unreal editor)
![launch error](https://user-images.githubusercontent.com/112169068/190849940-04a2ef32-4079-439b-9ddb-dd4a26408e93.PNG)

**Reply by ishaan95** [reply]
Did you specify UE4_ROOT environment variable? This is after following UE4 installation steps in documentation.

**Reply by arsalan0302** [reply]
Not actually, any suggestion how to specify UE4_ROOT environment variable?

**Reply by ishaan95** [reply]
Search "environment variables" on the Windows 10 search bar below and select "edit system environment variables", click environment variables button then and set it under the "user variables" category.
